### ISSUE TITLE
Illegal Data Address error

### ISSUE THREAD

**Issue by doctorfred** [ccutrer/waterfurnace_aurora#10]
Thank you for creating this!

My system is a Waterfurnace Series 7, IntelliZone 2, DHW, VS Drive, VS Pump

I built a cable as documented, using the DSD TECH SH-U11 adapter.

When starting Aurora MQTT bridge, I see an error:

```
Jan 26 16:10:52 raspberrypi aurora_mqtt_bridge[1276]: /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:283:in `query': The data address received in the query is not an allowable address for the server (ModBus::Errors::IllegalDataAddress)
Jan 26 16:10:52 raspberrypi aurora_mqtt_bridge[1276]:         from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:174:in `read_holding_registers'
Jan 26 16:10:52 raspberrypi aurora_mqtt_bridge[1276]:         from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:178:in `read_holding_register'
Jan 26 16:10:52 raspberrypi aurora_mqtt_bridge[1276]:         from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/proxy.rb:13:in `[]'
Jan 26 16:10:52 raspberrypi aurora_mqtt_bridge[1276]:         from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/lib/aurora/modbus/slave.rb:26:in `[]'
Jan 26 16:10:52 raspberrypi aurora_mqtt_bridge[1276]:         from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/lib/aurora/abc_client.rb:78:in `initialize'
Jan 26 16:10:52 raspberrypi aurora_mqtt_bridge[1276]:         from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/exe/aurora_mqtt_bridge:38:in `new'
Jan 26 16:10:52 raspberrypi aurora_mqtt_bridge[1276]:         from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/exe/aurora_mqtt_bridge:38:in `<top (required)>'
Jan 26 16:10:52 raspberrypi aurora_mqtt_bridge[1276]:         from /usr/local/bin/aurora_mqtt_bridge:23:in `load'
Jan 26 16:10:52 raspberrypi aurora_mqtt_bridge[1276]:         from /usr/local/bin/aurora_mqtt_bridge:23:in `<main>'
Jan 26 16:10:52 raspberrypi systemd[1]: aurora_mqtt_bridge.service: Main process exited, code=exited, status=1/FAILURE

```
Is this something specific to my system and firmware?  I get a different error if I unplug the cable from the ABC.  If I edit abc_client.rb and set the zone count to 2 on line 78, I get the same IllegalDataAddress later on:

```
Jan 26 16:31:20 raspberrypi systemd[1]: aurora_mqtt_bridge.service: Service RestartSec=3s expired, scheduling restart.
Jan 26 16:31:20 raspberrypi systemd[1]: aurora_mqtt_bridge.service: Scheduled restart job, restart counter is at 344.
Jan 26 16:31:20 raspberrypi systemd[1]: Stopped Aurora MQTT Bridge.
Jan 26 16:31:20 raspberrypi systemd[1]: Started Aurora MQTT Bridge.
Jan 26 16:31:24 raspberrypi aurora_mqtt_bridge[2500]: /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:283:in `query': The data address received in the query is not an allowable address for the server (ModBus::Errors::IllegalDataAddress)
Jan 26 16:31:24 raspberrypi aurora_mqtt_bridge[2500]:         from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/lib/aurora/modbus/slave.rb:12:in `read_multiple_holding_registers'
Jan 26 16:31:24 raspberrypi aurora_mqtt_bridge[2500]:         from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/lib/aurora/modbus/slave.rb:28:in `[]'
Jan 26 16:31:24 raspberrypi aurora_mqtt_bridge[2500]:         from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/lib/aurora/abc_client.rb:162:in `refresh'
Jan 26 16:31:24 raspberrypi aurora_mqtt_bridge[2500]:         from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/exe/aurora_mqtt_bridge:87:in `initialize'
Jan 26 16:31:24 raspberrypi aurora_mqtt_bridge[2500]:         from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/exe/aurora_mqtt_bridge:615:in `new'
Jan 26 16:31:24 raspberrypi aurora_mqtt_bridge[2500]:         from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/exe/aurora_mqtt_bridge:615:in `<top (required)>'
Jan 26 16:31:24 raspberrypi aurora_mqtt_bridge[2500]:         from /usr/local/bin/aurora_mqtt_bridge:23:in `load'
Jan 26 16:31:24 raspberrypi aurora_mqtt_bridge[2500]:         from /usr/local/bin/aurora_mqtt_bridge:23:in `<main>'
Jan 26 16:31:24 raspberrypi systemd[1]: aurora_mqtt_bridge.service: Main process exited, code=exited, status=1/FAILURE
Jan 26 16:31:24 raspberrypi systemd[1]: aurora_mqtt_bridge.service: Failed with result 'exit-code'.
```

**Comment by ccutrer** [comment]
Hmm... that's almost my exact configuration, so I'm surprised you would get that error. Can you do `aurora_fetch /dev/ttyUSB0 valid --yaml > myheatpump.yml` (adjusting the serial port if necessary) and then e-mail me the resulting file (assuming it doesn't run into errors)?

**Comment by doctorfred** [comment]
Same thing: 

aurora_fetch /dev/ttyUSB0 valid --yaml
Traceback (most recent call last):
	9: from /usr/local/bin/aurora_fetch:23:in `<main>'
	8: from /usr/local/bin/aurora_fetch:23:in `load'
	7: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/exe/aurora_fetch:35:in `<top (required)>'
	6: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/exe/aurora_fetch:35:in `new'
	5: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/lib/aurora/abc_client.rb:78:in `initialize'
	4: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.5/lib/aurora/modbus/slave.rb:26:in `[]'
	3: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/proxy.rb:13:in `[]'
	2: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:178:in `read_holding_register'
	1: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:174:in `read_holding_registers'
/var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:283:in `query': The data address received in the query is not an allowable address for the server (ModBus::Errors::IllegalDataAddress)

I’m wondering if there’s something wrong with the port or bit rate or some such.  I don’t know much of anything about RS-485/RS-422.

**Comment by ccutrer** [comment]
oh right, that makes sense... it goes through the same initialization first. lemme look and see if I can simplify that code to get past that part, and get to the part where it tries everything it can looking for valid registers

**Comment by ccutrer** [comment]
okay, 0.7.6 released that should bypass your error on initialization, and then try _really hard_ to get all the data it can.

**Comment by doctorfred** [comment]
This time it looks like it’s hanging on the read:

Successfully installed waterfurnace_aurora-0.7.6
2 gems installed
***@***.***:/usr/local/bin $ aurora_fetch /dev/ttyUSB0 valid
^CTraceback (most recent call last):
	25: from /usr/local/bin/aurora_fetch:23:in `<main>'
	24: from /usr/local/bin/aurora_fetch:23:in `load'
	23: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/exe/aurora_fetch:49:in `<top (required)>'
	22: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:64:in `query_registers'
	21: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:64:in `each'
	20: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:65:in `block in query_registers'
	19: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:71:in `rescue in block in query_registers'
	18: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:71:in `each'
	17: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:72:in `block (2 levels) in query_registers'
	16: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:77:in `rescue in block (2 levels) in query_registers'
	15: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:77:in `each'
	14: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:78:in `block (3 levels) in query_registers'
	13: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/modbus/slave.rb:26:in `[]'
	12: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/proxy.rb:13:in `[]'
	11: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:178:in `read_holding_register'
	10: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:174:in `read_holding_registers'
	 9: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:267:in `query'
	 8: from /usr/lib/ruby/2.5.0/timeout.rb:103:in `timeout'
	 7: from /usr/lib/ruby/2.5.0/timeout.rb:93:in `block in timeout'
	 6: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:269:in `block in query'
	 5: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/rtu_slave.rb:32:in `read_pdu'
	 4: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/modbus/slave.rb:35:in `read_rtu_response'
	 3: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/rtu.rb:43:in `read'
	 2: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/rtu.rb:43:in `loop'
	 1: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/rtu.rb:47:in `block in read'
/var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/rtu.rb:47:in `wait_readable': Interrupt

I hit ctrl-c after awhile.

**Comment by ccutrer** [comment]
It might take awhile. Possibly several minutes. You can also add `--no-try-individual` and it won't try _quite_ so hard, but might be missing some data.

**Comment by doctorfred** [comment]
I can see the send/receive light on the rs422 dogle going nuts and the send/receive on the ABC main board flashing i respose to your aurora_fetch.  I’ll wait awhile.

Thank you for helping!

**Comment by doctorfred** [comment]
Ok - it quit by itself:

***@***.***:/usr/local/bin $ aurora_fetch /dev/ttyUSB0 valid
Traceback (most recent call last):
	10: from /usr/local/bin/aurora_fetch:23:in `<main>'
	 9: from /usr/local/bin/aurora_fetch:23:in `load'
	 8: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/exe/aurora_fetch:49:in `<top (required)>'
	 7: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:64:in `query_registers'
	 6: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:64:in `each'
	 5: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:65:in `block in query_registers'
	 4: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:71:in `rescue in block in query_registers'
	 3: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:71:in `each'
	 2: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:72:in `block (2 levels) in query_registers'
	 1: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/modbus/slave.rb:14:in `read_multiple_holding_registers'
/var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:283:in `query': The function code received in the query is not an allowable action for the server (ModBus::Errors::IllegalFunction)


and

***@***.***:/usr/local/bin $ aurora_fetch /dev/ttyUSB0 valid --yaml --no-try-individual
Traceback (most recent call last):
	10: from /usr/local/bin/aurora_fetch:23:in `<main>'
	 9: from /usr/local/bin/aurora_fetch:23:in `load'
	 8: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/exe/aurora_fetch:49:in `<top (required)>'
	 7: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:64:in `query_registers'
	 6: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:64:in `each'
	 5: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:65:in `block in query_registers'
	 4: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:71:in `rescue in block in query_registers'
	 3: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:71:in `each'
	 2: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/abc_client.rb:72:in `block (2 levels) in query_registers'
	 1: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.6/lib/aurora/modbus/slave.rb:14:in `read_multiple_holding_registers'
/var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:283:in `query': The function code received in the query is not an allowable action for the server (ModBus::Errors::IllegalFunction)

was much faster, but same output.

**Comment by ccutrer** [comment]
0.7.7 will ignore that error as well and keep going.

**Comment by doctorfred** [comment]
—no-try-individual:

---
0: 0
1: 0
2: 204
3: 201
4: 32767
5: 240
6: 0
7: 0
8: 0
9: 120
10: 120
11: 0
12: 1
13: 0
14: 30
15: 30
16: 238
17: 400
18: 0
19: 662
20: 646
21: 1023
22: 0
23: 829
24: 1
25: 0
26: 32775
27: 589
28: 409
29: 15
30: 584
31: 400
32: 32767
33: 254
34: 60
35: 1
36: 2
37: 0
38: 0
39: 0
40: 0
41: 0
42: 65535
43: 0
44: 0
45: 0
46: 65535
47: 0
48: 0
49: 0
50: 2
51: 3
52: 10
53: 32767
54: 2
55: 65535
56: 65535
57: 65535
58: 65535
59: 65535
60: 65535
61: 65535
62: 65535
63: 65535
64: 65535
65: 65535
66: 65535
67: 0
68: 0
69: 0
70: 0
71: 6
72: 10
73: 653
74: 646
75: 283
76: 0
77: 0
78: 30
79: 0
80: 900
81: 5
82: 0
83: 829
84: 0
85: 0
86: 25
87: 14249
88: 16706
89: 17238
90: 21328
91: 8224
92: 20054
93: 22064
94: 13872
95: 16689
96: 12593
97: 17236
98: 21040
99: 19278
100: 8224
101: 8224
102: 8224
103: 8224
104: 8224
105: 12595
106: 12593
107: 12336
108: 13873
109: 13600
110: 0
111: 0
112: 240
113: 30
114: 120
115: 30
116: 65535
117: 65535
118: 120
119: 65535
120: 0
121: 3
122: 19535
123: 18208
124: 3
125: 21587
126: 21536
127: 1
128: 16719
129: 17184
130: 1
131: 19791
132: 17184
133: 1
134: 17733
135: 22048
136: 1
137: 16728
138: 16945
139: 3
140: 16728
141: 16946
142: 1
143: 18778
144: 12832
145: 65535
146: 65535
147: 65535
148: 65535
149: 65535
150: 0
151: 0
152: 0
153: 0
154: 0
155: 0
170: 0
171: 0
172: 0
173: 0
174: 41
175: 13890
176: 13107
177: 10293
178: 12334
179: 12592
180: 0
181: 0
182: 0
183: 0
184: 0
185: 0
186: 0
187: 57
188: 12334
189: 12592
190: 65535
191: 65535
192: 0
193: 0
194: 0
195: 0
196: 0
197: 32767
198: 0
199: 65535
200: 65535
201: 16732
202: 42691
203: 16732
204: 30902
205: 16793
206: 1466
207: 16778
208: 48491
209: 16792
210: 30301
211: 0
212: 0
213: 0
214: 0
215: 0
216: 0
217: 0
218: 0
219: 65535
220: 0
221: 0
222: 0
223: 0
224: 41
225: 14137
226: 14390
227: 10293
228: 12334
229: 12592
230: 0
231: 0
232: 0
233: 0
234: 0
235: 0
236: 0
237: 57
238: 12334
239: 12592
240: 65535
241: 65535
242: 0
243: 0
244: 0
245: 0
246: 0
247: 241
248: 16809
249: 47000
250: 17321
251: 47973
252: 0
253: 0
260: 136
280: 0
281: 26
282: 50
283: 207
284: 181
285: 191
286: 0
287: 0
288: 50
300: 0
301: 65535
340: 2
341: 3
342: 10
343: 32767
344: 2
345: 0
346: 65521
347: 11
348: 10
550: 18778
551: 12800
552: 0
553: 0
554: 0
555: 0
556: 0
557: 0
558: 0
559: 0
560: 108
561: 6
562: 0
563: 16
564: 0
565: 2
566: 0
567: 708
568: 0
569: 0
570: 0
571: 41
572: 50
573: 45
600: 0
601: 0
602: 0
603: 0
604: 6
605: 0
606: 65535
607: 10
608: 0
609: 65535
610: 65535
611: 0
612: 65535
613: 65535
614: 65535
615: 999
616: 0
617: 65535
618: 0
619: 3
620: 65535
621: 65535
622: 65535
623: 0
624: 0
625: 65535
626: 65535
627: 65535
628: 65535
629: 65535
630: 65535
631: 65535
632: 65535
633: 65535
634: 65535
635: 65535
636: 65535
637: 65535
638: 65535
639: 65535
640: 65535
641: 0
642: 0
643: 0
644: 0
645: 0
646: 0
647: 0
648: 0
649: 0
650: 65535
651: 0
652: 0
653: 0
654: 1
655: 20
656: 0
657: 20
658: 0
659: 3
660: 65535
661: 0
662: 65535
663: 65535
664: 65535
665: 65535
666: 65535
667: 65535
668: 65535
669: 65535
670: 65535
671: 0
672: 0
673: 0
674: 0
675: 65535
676: 65535
677: 65535
678: 65535
679: 65535
680: 65535
681: 65535
682: 65535
683: 65535
684: 65535
685: 65535
686: 65535
687: 65535
688: 65535
689: 65535
690: 65535
691: 65535
692: 65535
693: 65535
694: 65535
695: 65535
696: 65535
697: 65535
698: 65535
699: 65535
700: 0
1090: 16728
1091: 16979
1092: 21572
1093: 0
1094: 0
1095: 0
1096: 0
1097: 0
1098: 0
1099: 0
1100: 101
1101: 0
1102: 8
1103: 496
1104: 48
1105: 3
1106: 0
1107: 0
1108: 0
1109: 9999
1110: 623
1111: 615
1112: 55537
1113: 55537
1114: 756
1115: 1853
1116: 55537
1117: 0
1118: 0
1119: 0
1120: 0
1121: 99
1122: 0
1123: 0
1124: 9999
1125: 45538
1126: 0
1127: 0
1128: 95
1129: 3
1130: 0
1131: 0
1132: 0
1133: 0
1134: 652
1135: 9347
1136: 0
1137: 0
1138: 0
1139: 0
1140: 0
1141: 0
1142: 1
1143: 0
1144: 0
1145: 1
1146: 0
1147: 0
1148: 0
1149: 39
1150: 0
1151: 0
1152: 0
1153: 39
1154: 0
1155: 0
1156: 0
1157: 0
1158: 0
1159: 39
1160: 0
1161: 0
1162: 100
1163: 0
1164: 0
1165: 0
1200: 0
1201: 0
1202: 0
1203: 0
1204: 0
1205: 0
1206: 0
1207: 0
1208: 0
1209: 0
1210: 0
1211: 0
1212: 0
1213: 0
1214: 0
1215: 0
1216: 0
1217: 0
1218: 0
1219: 0
1220: 0
1221: 0
1222: 0
1223: 0
1224: 0
1225: 0
1226: 0
1227: 0
1228: 0
1229: 0
1230: 0
1231: 0
1232: 0
1233: 0
1234: 0
1235: 0
1236: 0
1237: 0
1238: 0
1239: 0
1240: 0
1241: 0
1242: 0
1243: 0
1244: 0
1245: 0
1246: 0
1247: 0
1248: 0
1249: 0
1250: 0
1251: 0
1252: 0
1253: 0
1254: 0
1255: 0
1256: 0
1257: 0
1258: 0
1259: 0
1260: 0
1261: 0
1262: 0
1263: 0
2000: 0
2001: 0
2002: 0
2003: 0
2004: 0
2005: 0
2006: 0
2007: 0
2008: 0
2009: 0
2010: 0
2011: 0
2012: 0
2013: 0
2014: 0
2015: 0
2016: 0
2017: 0
2018: 0
2019: 0
2020: 0
2021: 0
2022: 0
2023: 0
2024: 0
2025: 0
2026: 0
2100: 18778
2101: 12800
2102: 0
2103: 0
2104: 0
2105: 0
2106: 0
2107: 0
2108: 0
2109: 0
2110: 108
2111: 6
2112: 0
2113: 16
2114: 0
2115: 2
2116: 0
2117: 707
2118: 0
2119: 0
2120: 0
2121: 41
2122: 50
2123: 45
2124: 0
2125: 700
2126: 750
2127: 700
2128: 0
2129: 1
3000: 0
3001: 0
3002: 32767
3003: 0
3020: 0
3021: 0
3022: 0
3023: 0
3024: 2
3025: 0
3026: 0
3027: 0
3028: 16
3029: 0
3030: 0
3040: 32767
3041: 0
3042: 0
3043: 0
3044: 0
3045: 0
3046: 0
3047: 0
3048: 0
3049: 100
3060: 0
3061: 0
3062: 0
3063: 0
3100: 0
3101: 0
3102: 16403
3103: 13107
3104: 16492
3105: 52429
3108: 0
3109: 0
3110: 0
3111: 0
3112: 0
3113: 0
3114: 0
3115: 0
3118: 0
3119: 0
3200: 0
3201: 0
3202: 0
3203: 0
3204: 41
3205: 13890
3206: 13107
3207: 10293
3208: 12334
3209: 12592
3210: 0
3211: 0
3212: 0
3213: 0
3214: 0
3215: 0
3216: 0
3217: 57
3218: 12334
3219: 12592
3220: 0
3221: 0
3222: 0
3223: 0
3224: 0
3225: 0
3226: 0
3227: 0
3228: 0
3229: 2
3230: 2
3231: 16732
3232: 32874
3233: 16732
3234: 22046
3235: 0
3236: 16792
3237: 59711
3238: 16792
3239: 5559
3240: 16809
3241: 26492
3242: 1
3243: 0
3244: 0
3245: 0
3246: 0
3247: 4
3248: 16778
3249: 46462
3250: 17265
3251: 0
3252: 0
3253: 0
3300: 12337
3301: 11824
3302: 13608
3303: 13107
3304: 16950
3305: 10496
3306: 0
3307: 0
3308: 0
3309: 0
3310: 12337
3311: 11824
3312: 14592
3313: 0
3314: 0
3315: 0
3316: 0
3317: 0
3318: 0
3319: 0
3320: 105
3321: 109
3322: 1852
3323: 1850
3324: 0
3325: 664
3326: 662
3327: 701
3328: 0
3329: 0
3330: 620
3331: 241
3332: 0
3400: 0
3401: 0
3402: 0
3403: 0
3404: 41
3405: 14137
3406: 14390
3407: 10293
3408: 12334
3409: 12592
3410: 0
3411: 0
3412: 0
3413: 0
3414: 0
3415: 0
3416: 0
3417: 57
3418: 12334
3419: 12592
3420: 0
3421: 0
3422: 0
3423: 0
3424: 0
3425: 241
3426: 16809
3427: 47000
3428: 17321
3429: 50318
3430: 0
3431: 0
3500: 12337
3501: 11824
3502: 13608
3503: 13880
3504: 14647
3505: 10496
3506: 0
3507: 0
3508: 0
3509: 0
3510: 12337
3511: 11824
3512: 14592
3513: 0
3514: 0
3515: 0
3516: 0
3517: 0
3518: 0
3519: 0
3520: 105
3521: 109
3522: 702
3523: 340
3524: 0
3600: 0
3601: 0
3602: 0
3603: 0
3604: 0
3605: 0
3606: 0
3607: 0
3608: 0
3609: 0
3618: 0
3619: 0
3620: 0
3621: 0
3622: 0
3623: 0
3624: 0
3625: 0
3626: 0
3627: 0
3628: 0
3629: 0
3630: 0
3631: 0
3632: 0
3633: 0
3634: 0
3700: 0
3701: 0
3702: 0
3703: 0
3704: 0
3705: 0
3706: 0
3707: 0
3708: 0
3709: 0
3710: 0
3711: 0
3712: 0
3713: 0
3714: 0
3800: 64736
3801: 207
3802: 191
3803: 127
3804: 0
3805: 181
3806: 60
3807: 25
3808: 50
3809: 0
3818: 60
3819: 60
3820: 5
3821: 300
3822: 137
3823: 60
3824: 300
3825: 10
3826: 80
3827: 45
3828: 110
3829: 90
3830: 136
3831: 4
3832: 0
3833: 0
3834: 50
3900: 136
3901: 0
3902: 53786
3903: 693
3904: 664
3905: 646
3906: 45
3907: 108
3908: 108
3909: 9
3910: 1840
3911: 108
3912: 540
3913: 18
3914: 144
60050: 5900
60051: 5854
60052: 10230
60053: 8273
60100: 0
60101: 4
60102: 3
60103: 4
60104: 362
60105: 0
60106: 4
60107: 243
60108: 14584
60109: 35370
61000: 123
61001: 14424
61002: 36288
61003: 0
61004: 0
61005: 0
61006: 0
61007: 0
61008: 0
61009: 0

**Comment by doctorfred** [comment]
And w/o no-try-individual, it errors:

***@***.***:/usr/local/bin $ aurora_fetch /dev/ttyUSB0 valid --yaml > ~/out.yml
Traceback (most recent call last):
	17: from /usr/local/bin/aurora_fetch:23:in `<main>'
	16: from /usr/local/bin/aurora_fetch:23:in `load'
	15: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/exe/aurora_fetch:49:in `<top (required)>'
	14: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/lib/aurora/abc_client.rb:64:in `query_registers'
	13: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/lib/aurora/abc_client.rb:64:in `each'
	12: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/lib/aurora/abc_client.rb:65:in `block in query_registers'
	11: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/lib/aurora/abc_client.rb:71:in `rescue in block in query_registers'
	10: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/lib/aurora/abc_client.rb:71:in `each'
	 9: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/lib/aurora/abc_client.rb:72:in `block (2 levels) in query_registers'
	 8: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/lib/aurora/abc_client.rb:77:in `rescue in block (2 levels) in query_registers'
	 7: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/lib/aurora/abc_client.rb:77:in `each'
	 6: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/lib/aurora/abc_client.rb:78:in `block (3 levels) in query_registers'
	 5: from /var/lib/gems/2.5.0/gems/waterfurnace_aurora-0.7.7/lib/aurora/modbus/slave.rb:26:in `[]'
	 4: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/proxy.rb:13:in `[]'
	 3: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:178:in `read_holding_register'
	 2: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:174:in `read_holding_registers'
	 1: from /var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:266:in `query'
/var/lib/gems/2.5.0/gems/rmodbus-ccutrer-2.1.0/lib/rmodbus/client/slave.rb:275:in `rescue in query': Timed out during read attempt (ModBus::Errors::ModBusTimeout)

**Comment by ccutrer** [comment]
:nod: I'll try and look at the data tomorrow, see if I can make sense of it, and request more if I need to.

**Comment by doctorfred** [comment]
Problem was my system was at Firmware 2.0.4. Fixed in 0.8.0.  Thanks!
